app/src/ui/diff/text-diff.tsx

```diff
@@ -705,14 +705,13 @@ export class TextDiff extends React.Component<ITextDiffProps, {}> {
     if (canSelect(this.props.file)) {
       if (
         !canSelect(prevProps.file) ||
-        this.props.file.selection !== prevProps.file.selection
+        this.props.file.selection !== prevProps.file.selection ||
+        this.props.text === prevProps.text
       ) {
         // If the text has changed the gutters will be recreated
```

Comment:
It seems like this comment is potentially no longer correct?

Reply:
it's still correct! we do need to call `updateViewport()` when the text hasn't changed. The primary change in this PR is to call `updateViewport()` when the selection hasn't changed OR the text is the same. Previously it was AND, which means we missed some needed updates for the underlying code mirror instance.

Reply:
I'm going to make this logic a bit more readable in my next commit!